app/services/background.py

```
@@ -554,9 +557,258 @@ async def _market_data_sync_service(self):
+        Learns from:
+        - Active user positions
+        - Recent chat queries
+        - Trading volume patterns
+        - VIP user preferences
+
+        Returns prioritized list of symbols to cache.
+        """
+        intelligent_symbols = []
+        try:
+            # Method 1: Get symbols from active positions
+            async with AsyncSessionLocal() as db:
```

Ruff (0.13.2)

Undefined name `AsyncSessionLocal`

(F821)